### PR TITLE
OGM-1420 Test ElementCollection is loaded from JPQL query

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-1420
 

A couple of tests that I created some time ago.

### CODE REVIEW

File: core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+import java.util.List;
+
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.hibernate.ogm.utils.TestForIssue;
+import org.junit.After;
+import org.junit.Test;
+
+/**
+ * Test that the collection of elements is loaded with the entity when it's the result of a JPQL query using JPA.
```
Comment (fax4ever) [inline core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryTest.java]: What you think of: `Tests that {@link ElementCollection} is loaded within the entity when it's the result of a JPQL query using Hibernate.`

File: core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryWithJPATest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+import java.util.List;
+
+import org.hibernate.ogm.utils.TestForIssue;
+import org.hibernate.ogm.utils.jpa.OgmJpaTestCase;
+import org.junit.After;
+import org.junit.Test;
+
+/**
+ * Test that the collection of elements is loaded with the entity when it's the result of a JPQL query using Hibernat.
```
Comment (fax4ever) [inline core/src/test/java/org/hibernate/ogm/backendtck/embeddable/ElementCollectionFromJPQLQueryWithJPATest.java]: Even here, what do you think of `Tests that {@link ElementCollection} is loaded within the entity when it's the result of a JPQL query using JPA.'

**Comment by fax4ever** [comment]
Thank you @DavideD. Nice to have a test for it! I've just a very very minimal suggestion on test Javadoc.

**Comment by DavideD** [comment]
We can probably merge the two superclasses together

**Comment by fax4ever** [comment]
Merged #1058 thanks!

**Pull request closed without merging**